src/sdk/resources/diagnostics.ts

```diff
@@ -0,0 +1,11 @@
+import { IHttpClient } from '../http';
+import { EditableResource } from './resource';
+import { DiagnosticMetric } from '../models';
+
+export class DiagnosticMetricsResource extends EditableResource<DiagnosticMetric> {
```

Comment:
Metrics aren't editable. You should derive from `Resource` rather than `EditableResource`

Reply:
Actually, after looking at `Resource`, I'm not sure that is the right class either because it exposes a `get(id)` method, which isn't valid for metrics. I would actually refactor `Resource` into a `BaseResource` class that looks like this:

```javascript
export class BaseResource {
	public defaultParams: any;

	constructor(protected httpClient: IHttpClient) {}

	protected mergeDefaultParams(params: any): any {
		return this.mergeParams(this.defaultParams, params);
	}

	protected mergeParams(lhs: any, rhs: any): any {
		if (!lhs) {
			return rhs;
		}

		if (!rhs) {
			return lhs;
		}

		return Object.assign({}, lhs, rhs);
	}
}
```

and then this class becomes

```javascript
export class DiagnosticMetricsResource extends BaseResource {
	constructor(httpClient: IHttpClient) {
		super(httpClient);
	}

	async list(params?: any): Promise<TResource[]> {
		const result = await this.httpClient.get('diagnostics/metrics', this.mergeDefaultParams(params));
		return getTypedResponse<TResource[]>(result, 'metrics');
	}
}
```

Reply:
BaseResource class added to the Resources.ts, and updated the resources/diagnostics.ts to the suggested approach(with a few small tweaks).